src/Payments/Api.php

```diff
@@ -174,4 +175,58 @@ public function process_legacy_payment( PaymentContext $context, PaymentResult &
 		$result->set_payment_details( array_merge( $result->payment_details, $gateway_result ) );
 		$result->set_redirect_url( $gateway_result['redirect'] );
 	}
+
+	/**
+	 * Verify all dependencies of registered payment methods have been registered.
+	 * If not, remove that payment method script from the list of dependencies
+	 * of Cart and Checkout block scripts so it doesn't break the blocks and show
+	 * an error in the admin.
+	 */
+	public function verify_payment_methods_dependencies() {
+		global $wp_scripts;
+		$payment_method_scripts = $this->payment_method_registry->get_all_active_payment_method_script_dependencies();
+
+		foreach ( $payment_method_scripts as $payment_method_script ) {
+			$deps = $wp_scripts->registered[ $payment_method_script ]->deps;
+			foreach ( $deps as $dep ) {
+				if ( ! wp_script_is( $dep, 'registered' ) ) {
+					// Log and show errors.
+					error_log(  // phpcs:ignore
+						sprintf(
+							/* translators: 1: handle of the payment gateway. 2: handle of the dependency that is missing. */
+							esc_html__( 'Payment gateway with handle %1$s has been deactivated because its dependency %2$s is not registered.', 'woo-gutenberg-products-block' ),
```

Comment:
Do we need to localise debug strings? This is extra burden for translators, and we need to make sense of logs.

Reply:
Makes sense not localizing them. Changed in 23a26315014f8a4415194686889596c15721fbf3.